### PR TITLE
Feat/union filter

### PR DESCRIPTION
**Pull request by qingyashu**
This PR does:
  - Expand faceted search (union of fields instead of intersection)
  - Support range slider in faceted search
  - Add back numbers in filter
  - Fixed a bug in filtering data that couldn't match same arrays
  - Some improvements to the config for more convenient customization
  - Some necessary UI changes/improvs

Example to add more facets to the filter: 
```
"filterConfig": {
      "tabs": [{
        "title": "Subject",
        "fields": ["ethnicity", "gender", "species", "race"]
      },
      {
        "title": "Diagnosis",
        "fields": [
          "arthxbase",
          "bshbvstat",
          "bshcvstat",
          "cd4nadir",
          "status",
          "hiv_status"
        ]
      },
      {
        "title": "Drug Resistance",
        "fields": [
          "amikacin_res_phenotype",
          "capreomycin_res_phenotype",
          "isoniazid_res_phenotype",
          "kanamycin_res_phenotype",
          "ofloxacin_res_phenotype",
          "pyrazinamide_res_phenotype",
          "rifampicin_res_phenotype",
          "rifampin_res_phenotype",
          "streptomycin_res_phenotype"
        ]
      },
      {
        "title": "Resource",
        "fields": ["dataset"]
      }]
    },
```

### New Features
 - Expand faceted search (union of fields instead of intersection)

### Bug Fixes
  - Fixed a bug in filtering data that couldn't match same arrays

### Improvements
  - Add back numbers in filter
  - Some improvements to the config for more convenient customization, see "filterConfig" and "tableConfig" in gitops.json

### CODE REVIEW

File: src/Explorer/index.jsx [hunk 1]
```diff
@@ -1,68 +1,38 @@
...
     if (!row[property]) {
       return false;
     }
-    const filtersApplyMatch = filtersApplied[property].selectedValues.map(
-      x => x.toLowerCase(),
-    ) === row[property].toLowerCase();
-    let filtersApplyContains = filtersApplied[property].selectedValues.filter(
-      x => row[property].toLowerCase() === x.toLowerCase(),
+    let value = row[property];
+    if (typeof value === 'object' && value.length === 1) {
```
Comment (ZakirG) [inline src/Explorer/index.jsx]: What's going on here? I didn't think object-type variables had a length property defined

Comment (qingyashu) [reply src/Explorer/index.jsx]: it's trying to flatten an 1-length array, so previously array like ["white"] and "white" won't match

File: src/Explorer/index.jsx [hunk 2]
```diff
@@ -72,65 +42,69 @@ function checkIfFiltersApply(filtersApplied, row) {
...
+}
 
-  const result = {};
-  Object.keys(flattened).forEach((key) => {
-    result[key] = { histogram: [] };
-    Object.keys(flattened[key]).forEach((subKey) => {
-      result[key].histogram.push({ key: subKey, count: flattened[key][subKey] });
+function buildFilterTabsByCombinedAggsData(combinedAggsData) {
+  const result = config.dataExplorerConfig.filterConfig.tabs.map((t, i) => {
+    console.log(t);
```
Comment (ZakirG) [inline src/Explorer/index.jsx]: stray console :P

File: src/Explorer/index.jsx (hunk 1 above)
Comment (ZakirG) [reply src/Explorer/index.jsx]: hmm i'm confused -- then shouldn't it be `if (Array.isArray(value) && value.length === 1) ` ?

Comment (qingyashu) [reply src/Explorer/index.jsx]: Oh I didn't know Array has this `isArray` function, will change to this. Thanks! :)

File: src/Explorer/index.jsx [hunk 3]
```diff
@@ -217,14 +191,20 @@ class Explorer extends React.Component {
...
     const n = Object.keys(config.subcommons).length;
     for (let j = 0; j < n; j += 1) {
       promiseArray.push(
         this.obtainSubcommonsData(config.subcommons[j]),
       );
     }
+    promiseArray.push(
+      this.obtainSubcommonsData({
+        URL: '/',
+        name: 'Immport',
```
Comment (mfshao) [inline src/Explorer/index.jsx]:
```suggestion
        name: 'ImmPort',
```
this is picky but `ImmPort` is their official name

File: src/Explorer/index.jsx [hunk 4]
```diff
@@ -353,136 +327,86 @@ class Explorer extends React.Component {
...
     const n = Object.keys(config.subcommons).length;
     for (let j = 0; j < n; j += 1) {
       promiseArray.push(
         this.obtainSubcommonsAggsData(config.subcommons[j], filtersApplied),
       );
     }
+    promiseArray.push(
+      this.obtainSubcommonsAggsData({
+        URL: '/',
+        name: 'Immport',
```
Comment (mfshao) [inline src/Explorer/index.jsx]:
```suggestion
        name: 'Immport',
```
same here

File: src/Explorer/index.jsx (hunk 3 above)
Comment (qingyashu) [reply src/Explorer/index.jsx]: ooooo nice catch!

File: data/config/ndef.json [hunk 5]
```diff
@@ -237,17 +213,58 @@
...
+        "studyAccession",
+        "phenotype",
+        "gender",
+        "ethnicity",
+        "strain",
+        "species",
+        "submitter_id",
+        "race",
+        "hiv_status"
+      ]
```
Comment (ZakirG) [inline data/config/ndef.json]: Can you update the ndef-qa.json file accordingly as well? I'd like to deploy this PR to qa-test3 to test it

**Pull request merged**